plugins/jobs/src/js/JobDelete.js

```diff
@@ -0,0 +1,70 @@
+import * as React from "react";
+import { componentFromStream } from "data-service";
+import { deleteJob } from "#SRC/js/events/MetronomeClient";
+import { Subject } from "rxjs/Subject";
+import "rxjs/add/operator/map";
+import "rxjs/add/operator/switchMap";
+import "rxjs/add/operator/do";
+import "rxjs/add/operator/combineLatest";
+import "rxjs/add/operator/startWith";
+import "rxjs/add/operator/catch";
+
+import JobDeleteModal from "./components/JobDeleteModal";
+
+function errorIsTaskCurrentRunning(errorMsg) {
+  return /stopCurrentJobRuns=true/.test(errorMsg);
+}
+
+function executeDelete({ jobId, stopCurrentJobRuns, onSuccess, errorMsg }) {
+  return deleteJob(jobId, stopCurrentJobRuns)
+    .map(_ => ({ done: true, stopCurrentJobRuns, errorMsg }))
+    .do(_ => onSuccess())
+    .startWith({ done: false, stopCurrentJobRuns, errorMsg });
+}
+
+function deleteOperation() {
+  const deleteSubject$ = new Subject();
+  const delete$ = deleteSubject$.switchMap(executeDelete).catch(error => {
+    return delete$.startWith({
+      errorMsg: error.response.message,
```

Comment:
Are we sure `error.response.message` is always there?

Reply:
That is the only question that needs an action ☝️

Reply:
Yes. This comes from request, but I can guard against that.